docs/how_pool_works.md

```diff
@@ -0,0 +1,189 @@
+The receiver doesn't care, who exactly is sending on each block and in what configuration.
+The only thing that matters are the collectable per-cycle values below the timeline.
+This reduces the amount of stored data significantly, but this can be optimized further.
+
+![](how_pool_works_3.png)
+
+In this example, we started with having the raw collectable value of 23 for every block until
+the end of the blockchain.
+We then reduced that to storing values added to the collectable amount on each cycle.
+Now we need to describe only cycles when receiving anything.
+The senders usually are sending constant per-cycle amounts over long periods of time, so
+the added values tend to create long series of constant numbers, in this case, 5s.
+We exploit that and turn them into deltas relative to the previous cycles.
+Now we need deltas only for cycles where the sending rate changes, it's very cheap.
+This is what the contract actually stores: a mapping from cycle numbers to deltas.
+
+## Collecting
+
+When the receiver wants to collect the funds, they can iterate over the finished cycles, reconstruct
+the added values from deltas and then calculate the collectable amount at this point in time.
+For any cycle before the first funding begins the added value is assumed to be 0.
+
+# The sender
+
+The sender has a balance and a set of parameters, which describe the sending process.
+
+The balance can be increased by topping up, which requires sending the assets to the contract.
+The currently unspent balance can be withdrawn and transferred out of the contract at any time.
+
+The overall rate of funding of the sender is described with the amount per block.
+It limits the value, which can be taken from the balance on each block and sent to the receivers.
+This value stays constant over time unless explicitly updated.
+
+The sender maintains a list of receivers too, each of them with a weight.
+The weights regulate how the per-block amount is split between the receivers.
+For example, a receiver with weight 2 is going to get a share twice as big
+as a receiver with weight 1, but only half as big as another receiver with weight 4.
+
+## Starting sending
+
+When the sender has a sufficient balance, a sending rate and a list of receivers, sending starts.
+First, the funding period is calculated.
+Its start is the current block and its end is the block on which the balance will run out.
+Next, for each receiver the weighted share of the per-block rate is calculated.
+The receiver's deltas are updated to reflect that during the whole sending period on every block
+it's going to receive the calculated amount.
+
+Let's take a look at an example of an application of a delta.
+The sender will be sending 1 per block or 5 per cycle.
+
+![](how_pool_works_4.png)
+
+The deltas are applied relative to the existing values.
+It doesn't matter if anybody else is funding the receiver, it won't affect this sender.
```

Comment:
```suggestion
Note that this is a  timeline relative to a specific sender, therefore not considering possible other sends supporting this receiver.
```

Reply:
It's a timeline of a receiver, not a sender, it considers all the senders

Reply:
sorry, my bad. updated.

Reply:
But there's still only one timeline per receiver, the only relative thing are the changes, but that's already said in the previous sentence